### PR TITLE
Get single point surface datasets from subset_data rather than mksurfdata

### LINKED ISSUE #1674: Change mksurfdata_map/mksurfdata_esmf Makefile to build single-point datasets using subset_data
Currently mksurfdata.pl is used to create the single point surface datasets. We are moving that capability over to the new subset_data tool. So we need to change the Makefile in the mksurfdata_map/mksurfdata_esmf tool directory to use subset_data to build the single point datasets.

Relates to:

#1664

Blockers for this are: #1665 #1673

### LINKED ISSUE #1676: live coarse maintenance respiration is not included in the root respiration in src/biogeochem/CNVegCarbonFluxType.F90
### Brief summary of bug

[Live coarse maintenance respiration is not included in the root respiration in src/biogeochem/CNVegCarbonFluxType.F90. I am not sure whether it is a bug. I am just curious why based on the understanding of the root respiration. The fact that it is not included in the root respiration in the code doesn't agree with the description of plant maintenance respiration in the CLM Technical Note.]

### General bug information

**CTSM version you are using:** [Tag version PPE.n01_ctsm5.1.dev012.]

**Does this bug cause significantly incorrect results in the model's science?** [Not sure]

**Configurations affected:** [Fill this in if known.]

### Details of bug

[I am checking the respiration code in src/biogeochem/CNVegCarbonFluxType.F90
For the soil respiration starting at line 4547 in CNVegCarbonFluxType.F90
this%sr_col(c) = &
this%rr_col(c) + &
soilbiogeochem_hr_col(c)
where the root respiration is coded starting at line 4197:
this%rr_patch(p) = &
this%froot_mr_patch(p) + &
this%cpool_froot_gr_patch(p) + &
this%cpool_livecroot_gr_patch(p) + &
this%cpool_deadcroot_gr_patch(p) + &
this%transfer_froot_gr_patch(p) + &
this%transfer_livecroot_gr_patch(p) + &
this%transfer_deadcroot_gr_patch(p) + &
this%cpool_froot_storage_gr_patch(p) + &
this%cpool_livecroot_storage_gr_patch(p) + &
this%cpool_deadcroot_storage_gr_patch(p)

Why the live coarse root respiration livecroot_mr_patch(p) is not included in the calculation of root respiration as described in [the technical note](https://escomp.github.io/ctsm-docs/versions/master/html/tech_note/Plant_Respiration/CLM50_Tech_Note_Plant_Respiration.html) ?

I checked the size of each vegetation carbon pool (leaf, live stem, dead stem, fine root, live coarse root, dead coarse root) and found the live coarse root carbon pool is the smallest. I am wondering whether it is because the size of live coarse root carbon is much smaller compared to other carbon pools so that the maintenance respiration from the live coarse root carbon is negligible and not included in the root respiration calculation ? 

I don't have the respiration data from each carbon pool and couldn't check whether my assumption above is right. If someone else who has the data may go forward to check it. Or any other reason to explain the question ?]

### Important details of your setup / configuration so we can reproduce the bug
The code doesn't cause the model to abort.
### Important output or errors that show the problem

[Not sure whether it is a bug. If not, please explain the reason why the live coarse root maintenance respiration is not included in the root respiration in the code]

### LINKED ISSUE #1809: Add ability to name a different default config file for subset_data
I need the ability to name and use a different default config file for subset_data. This allows you to use subset data for example to use 1850 datasets, or a different resolution file.

I think eventually we plan to get the surface datasets from a fake_case that subset_data creates and that would supersede this usage. But, until we have that in place, this provides more flexibility that's easy to put into play.

This is required for #1674

### LINKED ISSUE #1924: Some updates to fsurdat_modifier script
There are some modifications I need to make to fsurdat_modifier tool in order to get it to work with 100% urban sites. See #1665
To do this in a general way I'm going to add some new sections to the config file for the script. I'll rename the main section from modify_inputs to modify_basic_options. The new sections will be: modify_subgrid_fractions, and modify_variable_list. modify_subgrid_fractions will allow you to override the PCT_* variables, and do error checking to make sure they are correct. modify_variable_list will allow you to modify an arbitrary list of variables that are on the file. Since it's an arbitrary list it won't have error checking, other than making sure it's a valid variable that can be set.

I am implementing in the context of a single point case, but it would work to set a regional or global case to constant values. And presumably it would work if you set the values in the config file to array values.

### LINKED ISSUE #1941: Add --silent option to ctsm_logging python infrastructure
I'd like to add a "--silent" option to ctsm_logging for our python scripts. This would allow an additional logging level that would suppress warnings, but still abort on error.

The reason this would be helpful is that subset_data writes out a warning if the --dom-pft option is used and for #1812 I'd like the warning to not show up.

### LINKED ISSUE #1942: Move py_env_create outside of tools test driver, as fails on compute nodes on cheyenne
### Brief summary of bug

When you send test_driver.sh to the compute nodes with qcmd, the call of py_env_create fails because you can't create a conda environment on the compute nodes. Hence, 

### General bug information

**CTSM version you are using:** ctsm5.1.dev115

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** test_driver.sh on cheyenne with qcmd

### Important output or errors that show the problem

```
Use conda to install the python environment needed to run the CTSM python tools in the conda environment: ctsm_py
Using the file: /glade/work/erik/ctsm_worktrees/answer_changes/python/conda_env_ctsm_py.txt
ctsm_py environment already exists
Install ctsm_py this can take a long time, be patient....
conda install --yes -q -c conda-forge -n ctsm_py --file /glade/work/erik/ctsm_worktrees/answer_changes/python/conda_env_ctsm_py.txt 
Collecting package metadata (current_repodata.json): ...working... failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://conda.anaconda.org/conda-forge/linux-64/current_repodata.json>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
'https//conda.anaconda.org/conda-forge/linux-64'


```

### PR DESCRIPTION
**Pull request by ekluzek**
### Description of changes

Get the single point surface datasets to be created using subset_data rather than mksurfdata.

### Specific notes

Contributors other than yourself, if any: @negin513 

CTSM Issues Fixed (include github issue #):
  Fixes #1676
  Fixes #1674
  Fixes #1809
  Fixes #1941 
  Fixes #1942
  Fixes #1924

Are answers expected to change (and if so in what way)? Yes (for single point sites)

Any User Interface Changes (namelist or namelist defaults changes)? Yes

Testing performed, if any: just running make for single point dataset files in tools/mksurfdata_map

### CODE REVIEW

**Comment by ekluzek** [comment]
@slevisconsulting I'm mostly done with the fsurdat_modifier part of this. Could you review that part of the code? I've got a little more to do, but it would really help for you to take a look and see if there are things I should change either in implementation or the testing for it.

**Comment by wwieder** [comment]
@ekluzek how much more still needs to come into this PR?  
- Do you just need a check from @slevisconsulting or are other reviews needed?  
- I realize there are additional changes to subset data coming in with #1932 that will also have to be updated before this PR can be merged, but that should be pretty straightforward?

File: python/ctsm/modify_input_files/fsurdat_modifier.py [hunk 1]
```diff
@@ -29,112 +29,382 @@ def main():
...
+    if dom_pft is not None:
+        modify_fsurdat.set_dom_pft(
+            dom_pft=dom_pft, lai=lai, sai=sai, hgt_top=hgt_top, hgt_bot=hgt_bot
+        )
+        logger.info("dom_pft complete")
+
+
+def read_opt_config(modify_fsurdat, config, cfg_path, section):
+    """read the optional parts of the main section config file"""
+    # not required: user may set these in the .cfg file
```
Comment (slevisconsulting) [inline python/ctsm/modify_input_files/fsurdat_modifier.py:292]: @ekluzek and I discussed this.
I propose renaming this function to show that it is operating on the optional stuff in the cfg file's first section only. The first section is `modify_fsurdat_basic_options`, so maybe name the function `read_cfg_optional_basic_opts`. 

For consistency, I suggest lengthening the docstring to say 
`Read the optional parts of the main section of the config file. The main section is called modify_fsurdat_basic_options. Users may set these optional parts but are not required to do so.`

This way the docstring incorporates the comment that's below it. If you update the docstring, then you can delete the separate comment.

**Review by slevisconsulting — APPROVED** [review]
@ekluzek thank you for meeting with me to go over the PR. The changes that we went over look good. From a compatibility perspective, I will discuss the few changes that @alperaltuntas has to worry about in the GUI:
1) The name of the main section of the cfg file: `modify_fsurdat_basic_options`
2) The two new boolean arguments in that section that will always be False for GUI applications:
a) process_subgrid_section = False
b) process_var_list_section = False

I proposed renaming a function for clarity of its purpose.

File: python/ctsm/modify_input_files/fsurdat_modifier.py (hunk 1 above)
Comment (ekluzek) [reply python/ctsm/modify_input_files/fsurdat_modifier.py]: OK, I did that change.

**Comment by ekluzek** [comment]
@wwieder this is almost done, and I'm planning on it coming in next. There are changes in possibly one or two of the same files as @adrifoster, but I don't think they will create conflicts. So merging them should be easy.

**Comment by ekluzek** [comment]
I went over this with @adrifoster and the two changes we thought would be helpful are to change "---file" in subset_data to something like "--cfg-file" so it's clear it's the config file and not the output or input file. The other change is to check for file existence before the RM in the Makefile so you don't see an error associated with something that's normal.

**Comment by ekluzek** [comment]
All testing is good other than the FSURDATMODIFYCTSM_D_Mmpi-serial_Ld1.5x5_amazon.I2000Clm50SpRs.cheyenne_intel test. There's an additional command line option that I need to give for it. I'm also working on it to make a failed test more obvious to debug.

I also found out that if you run once, it creates the output file and then aborts as it says the file is already created. So we should add an "--overwrite" option to overwrite the file if it already exists. I could add embedded code to remove the file if it exists, but the overwrite option would make it more obvious what it is doing.

**Comment by ekluzek** [comment]
OK, tests that are the same as ctsm5.1.dev115 the previous tag are the following 39:

- ERP_D_Ld5.f10_f10_mg37.I2000Clm50Vic.cheyenne_intel.clm-vrtlay			
- ERP_D_Ld5.f10_f10_mg37.IHistClm45Sp.cheyenne_intel.clm-decStart			
- ERP_Ld5.f10_f10_mg37.I2000Clm50Vic.cheyenne_gnu.clm-decStart			
- ERP_Ld9.f45_f45_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-FatesColdAllVars			
- ERP_P144x2_Ld30.f45_f45_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-mimicsFatesCold			
- ERP_P36x2_D.f10_f10_mg37.I2000Clm50SpRtmFl.cheyenne_intel.clm-default			
- ERP_P36x2_D_Ld5.f10_f10_mg37.I2000Clm45Sp.cheyenne_intel.clm-default			
- ERP_P36x2_Ld30.f45_f45_mg37.I2000Clm51FatesSpCruRsGs.cheyenne_intel.clm-FatesColdSatPhen			
- ERS_D_Ld3_PS.f09_g17.I2000Clm50FatesRs.cheyenne_intel.clm-FatesCold			
- ERS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm50FatesRs.cheyenne_intel.clm-FatesCold			
- ERS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm51FatesRs.cheyenne_intel.clm-FatesCold			
- ERS_Ld30.f45_f45_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-FatesColdFixedBiogeo			
- ERS_Ld30.f45_f45_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-FatesColdSizeAgeMort			
- ERS_Ld9.f10_f10_mg37.I2000Clm50FatesCruRsGs.cheyenne_intel.clm-FatesColdCH4Off			
- FSURDATMODIFYCTSM_D_Mmpi-serial_Ld1.5x5_amazon.I2000Clm50SpRs.cheyenne_intel			
- FUNITCTSM_P1x1.f10_f10_mg37.I2000Clm50Sp.cheyenne_intel			
- LILACSMOKE_D_Ld2.f10_f10_mg37.I2000Ctsm50NwpSpAsRs.cheyenne_intel.clm-lilac			
- PFS_Ld10_PS.f19_g17.I2000Clm50BgcCrop.cheyenne_intel			
- SMS.f45_f45_mg37.I2000Clm51FatesSpRsGs.cheyenne_nvhpc.clm-FatesColdSatPhen			
- SMS_D_Ld1_Mmpi-serial.f45_f45_mg37.I2000Clm50SpRs.cheyenne_intel.clm-ptsRLA			
- SMS_D_Ld5.f10_f10_mg37.I2000Clm45Fates.cheyenne_intel.clm-FatesCold			
- SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesCold			
- SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-FatesCold			
- SMS_D_Lm1_Mmpi-serial.CLM_USRDAT.I1PtClm50SpRs.cheyenne_intel.clm-USUMB_nuopc			
- SMS_D_Lm6_P144x1.f45_f45_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-FatesCold			
- SMS_D_Vmct_Lm1_Mmpi-serial.CLM_USRDAT.I1PtClm50SpRs.cheyenne_intel.clm-USUMB_mct			
- SMS_Ld1_PS.nldas2_rnldas2_mnldas2.I2000Ctsm50NwpSpNldasRs.cheyenne_gnu.clm-default			
- SMS_Ld5.f10_f10_mg37.I2000Clm45Fates.cheyenne_intel.clm-FatesCold			
- SMS_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-FatesCold			
- SMS_Ld5_PS.f19_g17.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesCold
- - ERS_Ld30.f45_f45_mg37.I2000Clm50FatesRs.izumi_intel.clm-FatesColdFixedBiogeo			
- SMS_D_Ld1_Mmpi-serial.f45_f45_mg37.I2000Clm50SpRs.izumi_gnu.clm-ptsRLA			
- SMS_D_Ld1_Mmpi-serial.f45_f45_mg37.I2000Clm50SpRs.izumi_gnu.clm-ptsROA			
- SMS_D_Ld1_Mmpi-serial.f45_f45_mg37.I2000Clm50SpRs.izumi_nag.clm-ptsRLA			
- SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.izumi_nag.clm-FatesCold			
- SMS_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Bgc.izumi_nag.clm-default--clm-NEON-NIWO			
- SMS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm50FatesRs.izumi_nag.clm-FatesCold			
- SMS_Ld5.f10_f10_mg37.I2000Clm45Fates.izumi_intel.clm-FatesCold			
- SMS_Ld5.f10_f10_mg37.I2000Clm50FatesRs.izumi_intel.clm-FatesCold

The are all short (the longest is 6-months). The single point cases aren't in the list as expected. Most are older physics as only the CLM51 cases will feel the zetamaxstable change.

There are only four CLM51 tests that show a difference in answers. I'm guessing the grids are small enough and they are run short enough that the zetamaxstable difference just doesn't happen to show up.

- ERP_P36x2_Ld30.f45_f45_mg37.I2000Clm51FatesSpCruRsGs.cheyenne_intel.clm-FatesColdSatPhen			
- ERS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm51FatesRs.cheyenne_intel.clm-FatesCold			
- SMS.f45_f45_mg37.I2000Clm51FatesSpRsGs.cheyenne_nvhpc.clm-FatesColdSatPhen			
- SMS_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Bgc.izumi_nag.clm-default--clm-NEON-NIWO


There's 35 total CLM51 tests, so 31 of them showed differences in answers.

**Pull request merged**